### PR TITLE
objectstore/cos: add `endpoint` field for config vpc internal endpoint

### PR DESCRIPTION
**Pull request by hanjm**
Signed-off-by: Jimmie Han <hanjinming@outlook.com>

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

1.  Add `endpoint` field to COS object store config.
2. Add document and validate test.
## Verification

1. Unit test.
2. Real depoyment test.

### CODE REVIEW

**Comment by hanjm** [comment]
CI is broken, it will fixed by https://github.com/thanos-io/thanos/pull/5000.

File: docs/storage.md [hunk 1]
```diff
@@ -419,6 +420,10 @@ config:
     max_conns_per_host: 0
 ```
 
+The `secret_key` and `secret_id` field is required. The `http_config` field is optional for optimize HTTP transport settings. There are two ways to configure the required bucket information:
+1. Provide the values of `bucket`, `region` and `app_id` keys.
+2. Provide the values of `endpoint` key with url format when you want to specific vpc internal endpoint. Please refer to the document of [endpoint](https://intl.cloud.tencent.com/document/product/436/6224) for more detail.
```
Comment (yeya24) [inline docs/storage.md:425]: specific -> specify.

**Review by yeya24 — COMMENTED** [review]
Looks super good!

File: docs/storage.md (hunk 1 above)
Comment (hanjm) [reply docs/storage.md]: Done. Thanks

**Review by yeya24 — APPROVED** [review]
LGTM

**Pull request merged**